clash-lib/src/Clash/Rewrite/Util.hs

```diff
@@ -201,9 +202,17 @@ apply = \s rewrite ctx expr0 -> do
 
   if lvl == DebugNone
     then return expr2
-    else applyDebug lvl dbgTranss fromLimit s expr0 hasChanged expr2
+    else do
+      Monad.when (lvl >= DebugCount && hasChanged) $ updateTransformCounters s
```

Comment:
Could we move this inside `applyDebug`, just so we have it all nice and cozy in one function?